Evolutionary_loop/Loop_Parts/Part_B/Part_B_PUEO_job2.sh

```diff
@@ -0,0 +1,116 @@
+#The list of frequencies, scaled up by 100 to avoid float operation errors in bash
+#we have to wait to change the frequencies since we're going to be changing them as we append them to simulation_PEC.xmacro (which is removed below before being remade)
+
+## We need the loop to pause until all the XF jobs are done
+## To do this, we'll just count the flag files
+
+cd $WorkingDir/Run_Outputs/$RunName/GPUFlags/
+flag_files=$(ls | wc -l) #$(ls -l --file-type | grep -v '/$' | wc -l)
+
+#Now we need to repeat that for the rest of the jobs
+while [[ $flag_files -lt $XFCOUNT ]] #we need to loop until flag_files reaches totPop
+do
+	sleep 1m
+	echo $flag_files
+	flag_files=$(ls | wc -l) #$(ls -l --file-type | grep -v '/$' | wc -l)
+done
+
+rm -f $WorkingDir/Run_Outputs/$RunName/GPUFlags/*
+
+echo $flag_files
+echo "Done!"
+
+# First, remove the old .xmacro files
+#when do that, we end up making the files only readable; we should just overwrite them
+#alternatively, we can just set them as rwe when the script makes them
+cd $XmacrosDir
+ 
+rm -f output.xmacro
+
+#echo "var m = $i;" >> output.xmacro
+echo "var NPOP = $NPOP;" >> output.xmacro
+echo "for (var k = $(($gen * $XFCOUNT + 1)); k <= $(($gen * $XFCOUNT + $XFCOUNT)); k++){" >> output.xmacro
+
+if [ $NSECTIONS -eq 1 ] # if 1, then the cone is symmetric
+then
+	cat shortened_outputmacroskeleton.txt >> output.xmacro
+else
+	cat shortened_outputmacroskeleton_Asym.txt >> output.xmacro
+fi
+
+
+sed -i "s+fileDirectory+${WorkingDir}+" output.xmacro
+# When we use the sed command, anything can be the delimiter between each of the arguments; usually, we use /, but since there are / in the thing we are trying to substitute in ($WorkingDir), we need to use a different delimiter that doesn't appear there
+
+
+module load xfdtd/7.9.2.2
+xfdtd $XFProj --execute-macro-script=$XmacrosDir/output.xmacro || true --splash=false
+
+#Xvnc :5 &  DISPLAY=:5 xfdtd $XFProj --execute-macro-script=$XmacrosDir/simulation_PEC.xmacro || true
+
+
+cd $WorkingDir/Antenna_Performance_Metric
+for i in `seq $(($gen*$XFCOUNT + $indiv)) $(($gen*$XFCOUNT+$XFCOUNT))`
+do
+	pop_ind_num=$(($i - $gen*$XFCOUNT))
+	for freq in `seq 1 131`
+	do
+		mv ${i}_${freq}.uan "$WorkingDir"/Run_Outputs/$RunName/${gen}_${pop_ind_num}_${freq}.uan
+	done
+done
+
+
+#I'm going to move the detector photos to the run directory
+cd $XmacrosDir
+chmod 775 *
+
+#FinalIndex=$(($NPOP-1))
+
+mkdir -m775 $WorkingDir/Run_Outputs/$RunName/Antenna_Images/${gen}
+for i in `seq $NPOP`
+do
+	mv ${i}_detector.png $WorkingDir/Run_Outputs/$RunName/Antenna_Images/${gen}/${i}_detector.png
+done
```

Comment:
```suggestion
mv *_detector.png $WorkingDir/Run_Outputs/$RunName/Antenna_Images/${gen}/
```
Can't you use something like this instead of a loop?

Reply:
I will ask for permission to change PUEO files in the group meeting.